photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala

```diff
@@ -54,4 +55,25 @@ class PalDBIndexMapLoader extends IndexMapLoader {
   override def indexMapForDriver(): IndexMap = new PalDBIndexMap().load(_storeDir, _numPartitions, _namespace)
 
   override def indexMapForRDD(): IndexMap = new PalDBIndexMap().load(_storeDir, _numPartitions, _namespace)
+
+  /**
+    * Converts a path object into a path string suitable for consumption by "sc.addFile". Implicitly converts a path
+    * with no specified scheme to the current default scheme (sc.addFiles doesn't do this automatically).
+    *
+    * @param path the input path
+    * @return the path string
+    */
+  protected[util] def getPath(sc: SparkContext, path: Path): String = {
+    var uri = path.toUri
+
+    Option(uri.getScheme) match {
+      case Some(scheme) => uri.toString
+      case _ =>
+        // If the path specifies no scheme, use the current default
+        val default = new Path(sc.hadoopConfiguration.get("fs.default.name")).toUri
+        new URI(
+          default.getScheme, default.getUserInfo, default.getHost, default.getPort,
+          uri.getPath, uri.getQuery, uri.getFragment).toString
```

Comment:
Scala style: If all of the arguments cannot fit on one line, they should each be on a separate line